Makefile

```diff
@@ -18,89 +18,51 @@
-
-TEST_PKG_LIST ?= $(shell go list ./... | grep -v github.com/tetratelabs/getenvoy/test/e2e)
-GO_TEST_OPTS ?=
-GO_TEST_EXTRA_OPTS ?=
-
-E2E_PKG_LIST ?= ./test/e2e
-# Run only one test at a time, in verbose mode, so that failures are easy to diagnose.
-# Note: -failfast helps as it stops at the first error. However, it is not a cacheable flag, so runs won't cache.
-E2E_OPTS ?= -parallel 1 -v -failfast
-E2E_EXTRA_OPTS ?=
-
-GOOSES := linux darwin
-GOARCHS := amd64
-
-GETENVOY_OUT_PATH = $(BIN_DIR)/$(1)/$(2)/getenvoy
-
-define GEN_GETENVOY_BUILD_TARGET
-.PHONY: $(call GETENVOY_OUT_PATH,$(1),$(2))
-$(call GETENVOY_OUT_PATH,$(1),$(2)):
-	CGO_ENABLED=0 GOOS=$(1) GOARCH=$(2) go build $(GO_LD_FLAGS) -o $(call GETENVOY_OUT_PATH,$(1),$(2)) ./cmd/getenvoy/main.go
-endef
-$(foreach os,$(GOOSES),$(foreach arch,$(GOARCHS),$(eval $(call GEN_GETENVOY_BUILD_TARGET,$(os),$(arch)))))
-
-.PHONY: deps
-deps:
-	go mod download
-
-.PHONY: build
-build: $(call GETENVOY_OUT_PATH,$(GOOS),$(GOARCH))
-
-.PHONY: release.dryrun
-release.dryrun:
-	goreleaser release --skip-publish --snapshot --rm-dist
+##@ Unit and End-to-End tests
 
+TEST_PACKAGES ?= $(shell go list ./... | grep -v github.com/tetratelabs/getenvoy/test/e2e)
 .PHONY: test
 test:
-	go test $(GO_TEST_OPTS) $(GO_TEST_EXTRA_OPTS) $(TEST_PKG_LIST)
+	@echo "--- test ---"
+	@go test $(TEST_PACKAGES)
 
+# End-to-end (e2e) tests run against a compiled binary. That dist is lazy built if "make dist" wasn't yet called.
+#
+# For e2e, run only one test at a time, in verbose mode, so that failures are easy to diagnose.
+# Note: -failfast helps as it stops at the first error. However, it is not a cacheable flag, so runs won't cache.
 .PHONY: e2e
-e2e: $(call GETENVOY_OUT_PATH,$(GOOS),$(GOARCH))
-	go test $(E2E_OPTS) $(E2E_EXTRA_OPTS) $(E2E_PKG_LIST)
-
-.PHONY: bin
-bin: $(foreach os,$(GOOSES), bin/$(os))
+e2e: $(BIN)
+	@echo "--- e2e ---"
+	@go test -parallel 1 -v -failfast ./test/e2e
 
-define GEN_BIN_GOOS_TARGET
-.PHONY: bin/$(1)
-bin/$(1): $(foreach arch,$(GOARCHS), bin/$(1)/$(arch))
-endef
-$(foreach os,$(GOOSES),$(eval $(call GEN_BIN_GOOS_TARGET,$(os))))
-
-define GEN_BIN_GOOS_GOARCH_TARGET
-.PHONY: bin/$(1)/$(2)
-bin/$(1)/$(2): $(call GETENVOY_OUT_PATH,$(1),$(2))
-endef
-$(foreach os,$(GOOSES),$(foreach arch,$(GOARCHS),$(eval $(call GEN_BIN_GOOS_GOARCH_TARGET,$(os),$(arch)))))
+##@ Code quality and integrity
 
-COVERAGE_PACKAGES ?= $(shell echo $(TEST_PKG_LIST)| tr -s " " ",")
+COVERAGE_PACKAGES ?= $(shell echo $(TEST_PACKAGES)| tr -s " " ",")
 .PHONY: coverage
 coverage:
 	@echo "--- coverage ---"
-	@rm -rf coverage
-	@mkdir coverage
-	@go test -coverprofile=coverage/coverage.txt -covermode=atomic --coverpkg $(COVERAGE_PACKAGES) $(TEST_PKG_LIST)
-	@go tool cover -html=coverage/coverage.txt -o coverage/coverage.html
-
-##@ Code quality and integrity
+	@go test -coverprofile=coverage.txt -covermode=atomic --coverpkg $(COVERAGE_PACKAGES) $(TEST_PACKAGES)
+	@go tool cover -func coverage.txt
```

Comment:
since we now use codecov.io for visualizing in html, this tool is default (which aggregates whereas go's html does not)